mailpoet/lib/Automation/Engine/Validation/WorkflowRules/ValidStepRule.php

```diff
@@ -63,6 +69,61 @@ public function visitNode(Workflow $workflow, WorkflowNode $node): void {
     }
   }
 
+  private function mapErrorCodesToErrorMessages(array $errorCodes): array {
+
+    return array_map(
+      function(string $errorCode): string {
+        switch ($errorCode) {
+          case "rest_property_required":
+            return __('This is a required field.', 'mailpoet');
+          case "rest_additional_properties_forbidden":
+            return "";
+          case "rest_too_few_properties":
+            return "";
+          case "rest_too_many_properties":
+            return "";
+          case "rest_invalid_type":
+            return __('This field is not well formed.', 'mailpoet');
+          case "rest_too_few_items":
+            return __('Please add more items.', 'mailpoet');
+          case "rest_too_many_items":
+            return __('Please remove some items.', 'mailpoet');
+          case "rest_duplicate_items":
+            return __('Please remove duplicate items.', 'mailpoet');
+          case "rest_invalid_multiple":
+            return __('This field is not well formed.', 'mailpoet');
+          case "rest_out_of_bounds":
+            return __('This value is out of bounds.', 'mailpoet');
+          case "rest_too_short":
+            return __('This value is not long enough.', 'mailpoet');
+          case "rest_too_long":
+            return __('This value is too long.', 'mailpoet');
+          case "rest_invalid_pattern":
+            return __('This value is not well formed.', 'mailpoet');
+          case "rest_no_matching_schema":
+            return __('This value does not match the expected format.', 'mailpoet');
+          case "rest_one_of_multiple_matches":
+            return __('This value is not matching the correct times.', 'mailpoet');
+          case "rest_not_in_enum":
+            return __('This value is not well formed.', 'mailpoet');
+          case "rest_invalid_hex_color":
+            return __('This value is not a hex formatted color.', 'mailpoet');
+          case "rest_invalid_date":
+            return __('This value is not a date.', 'mailpoet');
+          case "rest_invalid_email":
+            return __('This value is not an email address.', 'mailpoet');
+          case "rest_invalid_ip":
+            return __('This value is not an IP address.', 'mailpoet');
+          case "rest_invalid_uuid":
+            return __('This value is not an UUID.', 'mailpoet');
+          default:
+            return $errorCode;
+        }
+      },
```

Comment:
I was thinking... can we improve the existing messages just by replacing some substrings? But then I realized nope, because it would not be translatable. But... this doesn't seem to cover things like `value must be at most 3 characters long.`. It also sets some messages to empty strings.

The validator is a carefully designed WP schema wrapper, and we must be very cautious not to bend it to something else just because of a specific feature. It relates to my previous comments about whether we should use it from the outside for now, maybe it would be safer.